Src/IronPython/Runtime/Operations/IntOps.cs

```diff
@@ -540,19 +540,16 @@ public static partial class Int32Ops {
             if (bytesArr.Length == 0) return 0;
 
 #if NETCOREAPP
-            return new BigInteger(bytesArr.AsSpan(), isUnsigned: !signed, isBigEndian: !isLittle);
+            var val = new BigInteger(bytesArr.AsSpan(), isUnsigned: !signed, isBigEndian: !isLittle);
 #else
 
-            if (isLittle) {
-                bool msbSet = (bytesArr[bytesArr.Length - 1] & 0x80) == 0x80;
-                if (!msbSet) return new BigInteger(bytesArr);
-                return new BigInteger(bytesArr.Concat(Enumerable.Repeat<byte>(signed ? (byte)0xff : (byte)0, 1)).ToArray());
-            } else {
-                bool msbSet = (bytesArr[0] & 0x80) == 0x80;
-                if (!msbSet) return new BigInteger(bytesArr.Reverse());
-                return new BigInteger(bytesArr.Reverse().Concat(Enumerable.Repeat<byte>(signed ? (byte)0xff : (byte)0, 1)).ToArray());
-            }
+            if (!isLittle) bytesArr = bytesArr.Reverse();
+            bool msbSet = (bytesArr[bytesArr.Length - 1] & 0x80) == 0x80;
+            var val = msbSet
+                ? new BigInteger(bytesArr.Concat(Enumerable.Repeat<byte>(signed ? (byte)0xff : (byte)0, 1)).ToArray())
+                : new BigInteger(bytesArr);
```

Comment:
I think if `signed` then there is no need for padding with `0xff` because `msbSet` so the value will already be recognized as negative. Something like:
```C#
var val = msbSet && !signed
    ? new BigInteger(bytesArr.Concat(Enumerable.Repeat<byte>((byte)0, 1).ToArray())
    : new BigInteger(bytesArr);
```
?
Not tested.